### PR TITLE
[OpenVINO] Fix array(), maximum(), minimum(), reshape() and split() ops in the numpy suite

### LINKED ISSUE #34409: [Keras 3 OpenVINO backend]: Improve support for array(), maximum(), minimum(), reshape() and split() operations
### Request Description

This issue tracks improving support for `array(), maximum(), minimum(), reshape() and split()` ops in the numpy suite of the Keras OpenVINO backend.

The operations is currently not implemented properly in keras/backend/openvino, and have some tests excluded, limiting NumPy parity for this backend. 

I will submit a PR implementing this and enable the corresponding tests.

Parent Issue: #34017

### Feature Use Case

_No response_

### Issue submission checklist

- [x] The feature request or improvement must be related to OpenVINO

### PR DESCRIPTION
**Pull request by goyaladitya05**
This PR focuses on improving the implementations for `array(), maximum(), minimum(), reshape() and split()` for the OpenVINO Backend in the numpy suite.

- array() - replaced `np.array()` with `convert_to_tensor()` so the return value is always an `OpenVINOKerasTensor`, consistent with other ops.

- maximum() / minimum() - added JAX weak-type dtype promotion so Python int/float literals are promoted correctly. The shared logic was extracted into a `_promote_binary_types()` helper, removing duplication across add(), subtract(), multiply() too.

- reshape() - fixed three cases: bare int newshape is wrapped in a list (OpenVINO requires a 1-D shape tensor); None dims are replaced with -1; an `OpenVINOKerasTensor` newshape (side-effect of the array() fix) is handled via get_ov_output().

- split() - handles `OpenVINOKerasTensor` as indices_or_sections by extracting constant data via `.output.get_node().get_data()`; side-effect of the array() fix, also covers `hsplit() / vsplit()`.

- convert_to_numpy() - wrapped read-only return paths in np.array() to guarantee a writable copy, mirroring the fix already applied to the JAX and TensorFlow backends.

### Duplicate of #22319 
The CI was getting stuck, so I looked deeper and found the main issue. 
OpenVINO tensors rely on sequence style iteration via` __getitem__`, but `__getitem__` never raises `IndexError`, so loops like `zip(mask, mask2)` become unbounded and CI appears stuck. I’m adding a bounded __iter__ on OpenVINOKerasTensor to make iteration finite and deterministic.

Closes: openvinotoolkit/openvino/issues/34409

### CODE REVIEW

File: keras/src/backend/openvino/numpy.py [hunk 1]
```diff
@@ -3469,8 +3448,12 @@ def repeat(x, repeats, axis=None):
...
     x = get_ov_output(x)
-    if isinstance(newshape, tuple):
+    if isinstance(newshape, int):
+        newshape = [newshape]
+    elif isinstance(newshape, tuple):
         newshape = list(newshape)
+    # Replace None (dynamic dims) with -1, which OpenVINO uses for inference
+    newshape = [-1 if d is None else d for d in newshape]
     newshape = ov_opset.constant(newshape, Type.i32).output(0)
     return OpenVINOKerasTensor(ov_opset.reshape(x, newshape, False).output(0))
```
[the remark this answers is not in the document]
Comment (goyaladitya05) [reply keras/src/backend/openvino/numpy.py:3493]: done

**Comment by goyaladitya05** [comment]
I've rebased the PR, and all tests are passing. Please review. Thanks!

**Comment by hertschuh** [comment]
> Duplicate of https://github.com/keras-team/keras/pull/22319

Closing since the other one was submitted. Please re-open if this is not the case.

**Comment by goyaladitya05** [comment]
> Closing since the other one was submitted. Please re-open if this is not the case.

We merged it in #22319, then we reverted that submission in #22342, since the CI was getting stuck. So effectively, no changes were merged earlier.

**Comment by goyaladitya05** [comment]
@hertschuh Could you reopen this please? I'm not getting an option to reopen it myself.

**Comment by hertschuh** [comment]
Done. Please rebase.

**Comment by goyaladitya05** [comment]
@hertschuh This PR can be reviewed now, thanks.

File: keras/src/backend/openvino/numpy.py [hunk 2]
```diff
@@ -3671,6 +3698,11 @@ def split(x, indices_or_sections, axis=0):
         shape_tensor, axis_i32, ov_opset.constant(0, dtype=Type.i32)
     )
 
+    if isinstance(indices_or_sections, OpenVINOKerasTensor):
+        indices_or_sections = (
+            indices_or_sections.output.get_node().get_data().tolist()
+        )
```
[the remark this answers is not in the document]
Comment (goyaladitya05) [reply keras/src/backend/openvino/numpy.py]: Actually no, the earlier would be better.

**Review by hertschuh — APPROVED** [review]
Thanks for the fixes!
